ncolony/ctllib.py

```diff
@@ -39,23 +39,81 @@
 import json
 import os
 
+import attr
 from twisted.python import filepath
 
 from ncolony import main as mainlib
 
 NEXT = functools.partial(next, itertools.count(0))
 
-Places = collections.namedtuple('Places', 'config messages')
-
+@attr.s(frozen=True)
+class Places(object):
+    config = attr.ib()
+    messages = attr.ib()
 
 def _dumps(stuff):
     return json.dumps(stuff).encode('utf-8')
 
+_COMMAND_LINE = object()
+
+def attrib(tp=str, convert=None, required=False, positional=False):
+    metadata = {_COMMAND_LINE: (tp, positional, convert)}
+    default = attr.NOTHING
+    if not required:
+        if type == dict:
```

Comment:
where is type coming from? i think this type is the builtin `type` which would always evalulate to false here. i tink you mean to use `tp`?